examples/design/EU-DEMO/EUDEMO_builders/blanket.py

```diff
@@ -0,0 +1,188 @@
+        ibs.plot_options.face_options["color"] = BLUE_PALETTE[self.BB][0]
+        obs = PhysicalComponent(self.OBS, obs_silhouette)
+        obs.plot_options.face_options["color"] = BLUE_PALETTE[self.BB][1]
+        return Component(self.BB, children=[ibs, obs])
+
+    def build_xy(self, segments: List[PhysicalComponent]):
+        """
+        Build the x-y components of the blanket.
+        """
+        xy_plane = BluemiraPlacement.from_3_points([0, 0, 0], [1, 0, 0], [0, 1, 0])
+
+        slices = []
+        for i, segment in enumerate(segments):
+            single_slice = PhysicalComponent(
+                segment.name, BluemiraFace(slice_shape(segment.shape, xy_plane)[0])
+            )
+            single_slice.plot_options.face_options["color"] = BLUE_PALETTE[self.BB][i]
+            slices.append(single_slice)
+
+        return circular_pattern_component(
+            Component(self.BB, children=slices), self.params.n_TF.value
+        )
+
+    def build_xyz(self, segments: List[PhysicalComponent], degree: float = 360.0):
+        """
+        Build the x-y-z components of the blanket.
+        """
+        sector_degree, n_sectors = get_n_sectors(self.params.n_TF.value, degree)
+
+        # TODO: Add blanket cuts properly in 3-D
+        return circular_pattern_component(
+            Component(self.BB, children=segments),
+            n_sectors,
+            degree=sector_degree * n_sectors,
+        )
+
+    def get_segments(self, ibs_silhouette: BluemiraFace, obs_silhouette: BluemiraFace):
+        """
+        Create segments of the blanket from inboard and outboard silhouettes
+        """
+        ibs_shapes = pattern_revolved_silhouette(
+            ibs_silhouette,
+            self.params.n_bb_inboard.value,
+            self.params.n_TF.value,
+            self.params.c_rm.value,
+        )
+
+        obs_shapes = pattern_revolved_silhouette(
+            obs_silhouette,
+            self.params.n_bb_outboard.value,
+            self.params.n_TF.value,
+            self.params.c_rm.value,
+        )
+
+        segments = []
+        for name, base_no, bs_shape in [
+            [self.IBS, 0, ibs_shapes],
+            [self.OBS, self.params.n_bb_inboard.value + 1, obs_shapes],
+        ]:
+            for no, shape in enumerate(bs_shape):
+                segment = PhysicalComponent(f"{name}_{no}", shape)
+                segment.display_cad_options.color = BLUE_PALETTE[self.BB][base_no + no]
+                segments.append(segment)
+        return segments
+
+    def segment_blanket(self, offset: float = 0.1):
+        """
+        Cut the blanket silhouette into segment silhouettes. Simple vertical cut for now.
+        """
+        bb = self.silhouette.bounding_box
+        x_mid = 0.5 * (bb.x_min + bb.x_max)
+        delta = 0.5 * self.params.c_rm.value
+
+        x = np.full(4, x_mid)
+        x[
+            (0, 3),
+        ] -= delta
+        x[
+            (1, 2),
+        ] += delta
```

Comment:
Good job, black 😆. Would removing the trailing comma improve the formatting here?

Reply:
The indexing then moans that I'm indexing too axes. I will have a play